Troubleshooting.md

```diff
@@ -393,3 +396,11 @@ For backward compatibility it is possible to restore Gson's old behavior of allo
 
 - This does not solve any of the type-safety problems mentioned above; in the long term you should prefer one of the other solutions listed above. This system property might be removed in future Gson versions.
 - You should only ever set the property to `"true"`, but never to any other value or manually clear it. Otherwise this might counteract any libraries you are using which might have deliberately set the system property because they rely on its behavior.
+
+## R8 / ProGuard
```

Comment:
I didn't apply `## <a id="r8-proguard"></a> R8 / ProGuard` for this header like the others, as the index of it (`#a-idr8-proguarda-r8--proguard`) can't be honored by GitHub web correctly.

Reply:
Is that specific to this header? Because for the others GitHub UI seems to ignore the `<a>` when deriving the default anchor name (or it only seems to prepend a `-`).

If possible I think a stable anchor with `<a>` would be better.

Reply:
Seems all existing ones are invalid.

Reply:
View https://github.com/Goooler/gson/blob/e8f52fb490dc7b58d95c01447701b986e655a214/Troubleshooting.md#a-idproguard-r8a-proguard--r8.

The ref id works in IDEA's markdown render but GitHub website.

Reply:
That rather looks like an issue with IntelliJ. In GitHub both anchors work:
- GitHub derived one (also used for the "🔗" icon next to the title):
[`#-proguard--r8`](https://github.com/Goooler/gson/blob/e8f52fb490dc7b58d95c01447701b986e655a214/Troubleshooting.md#-proguard--r8)
- The one assigned with `<a>` (which should be referred to within the guide)
[`#proguard-r8`](https://github.com/Goooler/gson/blob/e8f52fb490dc7b58d95c01447701b986e655a214/Troubleshooting.md#proguard-r8)

(though maybe for GitHub this also relies on implementation details to some extent)

Reply:
I prefer to keep the header without id, it works for both IDEA and GitHub. But I can update the style if you want.

Reply:
IntelliJ (IntelliJ IDEA 2024.3.5) seems to be a bit inconsistent. Code completion for Markdown links suggests these `#a-idr8-proguarda-r8--proguard` anchors you mentioned. However, for the Markdown preview the custom `<a>` anchor links seem to work fine within the same document. But they do not work between different files, e.g. the link from the README to the Troubleshooting guide section.

Reply:
Yeah, the same behavior as IDEA 2025.1